.github/workflows/pr-preview.yml

```diff
@@ -0,0 +1,99 @@
+name: PR Preview
+
+on:
+  push:
+    branches:
+      - '**'
```

Comment:
Pull requests from forks should not trigger this workflow, as it may leak a secret. By default, pull requests from forks do not have access to secrets, but there are some dangerous ways to get around this. 

If we want to do it secure so that unauthorized people do not gain access to our secret, we should divide this workflow into 2 workflows: 
- The unprivileged(`pull_request`): Executes the user code and publishes the build result as artifacts. This workflow does not have access to the secret.
- The privileged(`workflow_run`): Only executes code from the main branch for our repository, and has access to secrets and artifacts from other builds.

Here's a example repository that implements this pattern:
https://github.com/dependabot/dependabot-actions-workflow

Reply:
Ah, okay. I assumed that it'd be safe to rely on GitHub to keep secrets from being leaked, but it's probably better to be extra-cautious about this.

In that case, I think it'd probably be fine to just restrict this workflow given that PR previews aren't used extremely frequently, so pushing a branch upstream rather than on your own fork can be done whenever a preview is desired.